relayer/relays/beacon/header/header.go

```diff
@@ -487,14 +403,13 @@ func (h *Header) getClosestCheckpoint(slot uint64) (cache.Proof, error) {
 	checkpoint, err := h.cache.GetClosestCheckpoint(slot)
 
 	switch {
-	case errors.Is(cache.FinalizedCheckPointNotAvailable, err):
-		calculatedCheckpointSlot := h.syncer.CalculateNextCheckpointSlot(slot)
-
-		log.WithFields(log.Fields{"calculatedCheckpointSlot": calculatedCheckpointSlot}).Info("checkpoint slot not available")
-
-		return cache.Proof{}, err
-	case errors.Is(cache.FinalizedCheckPointNotPopulated, err):
-		err := h.populateFinalizedCheckpoint(checkpoint.Slot)
+	case errors.Is(cache.FinalizedCheckPointNotAvailable, err) || errors.Is(cache.FinalizedCheckPointNotPopulated, err):
+		checkpointSlot := checkpoint.Slot
+		if checkpointSlot == 0 {
+			checkpointSlot = h.syncer.CalculateNextCheckpointSlot(slot)
+			log.WithFields(log.Fields{"calculatedCheckpointSlot": checkpointSlot}).Info("checkpoint slot not available")
+		}
+		err := h.populateFinalizedCheckpoint(checkpointSlot)
```

Comment:
Merge these two branches together and `populateFinalizedCheckpoint` only if exists on chain.